File: BinaryDecimal.py
def binary_to_decimal(decimal : int) -> list[int]:
    if decimal == 0: return []
    res = []
    while decimal != 0:
        res.append(decimal % 2)
        decimal = int(decimal / 2) # decimal //= 2
    res.reverse()
    return res

def decimal_to_binary(binary : list[int]) -> int:
    return sum(b * (2 ** i) for i, b in enumerate(reversed(binary)))

def make_n(n: int, binary: list[int]) -> list[int]:
    res = binary[:]
    if n > len(binary):
        # Append trailing zeros
        res.extend([0] * (n - len(binary)))
    else:
        # Truncate to the first n elements
        res = binary[:n]
    
    return res
    
    return res
print(binary_to_decimal(13))
print(decimal_to_binary(binary_to_decimal(13)))
print(make_n(8, [1,0,1,1,0,0,1,1,1,1]))

File: BinaryDecimal.cc
#include <vector>
#include <iostream>

std::vector<int> decimal_to_binary(int decimal)
{
    if (decimal == 0)
        return {0};

    std::vector<int> results{};
    while (decimal != 0)
    {
        results.push_back(decimal % 2);
        decimal /= 2;
    }
    std::reverse(results.begin(), results.end());

    return results;
}

int binary_to_decimal(std::vector<int> binary)
{
    int res = 0;
    int n = binary.size();
    for (int i = 0; i < n; ++i)
    {
        res += binary[n - 1 - i] * std::pow(2, i);
    }
    return res;
}

int main(void)
{
    std::vector<int> res = decimal_to_binary (13);
    for (const int& elem : res){
        std::cout<<elem<<'\t';
    }
    return 0;
}